src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -456,6 +467,58 @@ private void putNLPResultToSourceMapForMapType(
         }
     }
 
+    /**
+     * Put nlp result to single source element, which is in a list field of source document
+     * Such source element is in map type
+     *
+     * @param processorKey
+     * @param sourceValue
+     * @param results
+     * @param indexWrapper
+     * @param sourceAndMetadataMap
+     * @param nestedElementIndex index of the element in the list field of source document
+     */
+    @SuppressWarnings("unchecked")
+    private void putNLPResultToSingleSourceMapInList(
+        String processorKey,
+        Object sourceValue,
+        List<?> results,
+        IndexWrapper indexWrapper,
+        Map<String, Object> sourceAndMetadataMap,
+        int nestedElementIndex
+    ) {
+        if (processorKey == null || sourceAndMetadataMap == null || sourceValue == null) return;
+        if (sourceValue instanceof Map) {
```

Comment:
Isn't sourceValue always an instance of Map?

Reply:
No. In the case of doc with list of nested objects, sourceValue will become type of List in the last recursive call. You may check line 505 - 506

Reply:
I mean, this method is called only when the sourceValue is instance of Map.

Reply:
Yes, it is called in `putNLPResultToSourceMapForMapType()` when `sourceValue` is Map, but `sourceValue` is a nested object with `list` inside. During the recursive call of `putNLPResultToSingleSourceMapInList()`, it may reach to the level of `list` type

Reply:
You're right. I missed the recursive aspect. I have a follow-up comment:  

Is there a limit on the number of recursive calls that can occur? There was a security campaign emphasizing the need to avoid unlimited recursive calls. We should either impose a limit on the recursion depth or refactor the logic to use an iterative approach.

Reply:
Yes. We have done `fieldMap` depth validation in `ProcessorDocumentUtils.validateMapTypeValue()`. Given the recursion depth of this method is same as `fieldMap` depth, it should be fine for us to not do recursion depth validation again here. 

Pleas feel free to let me know your thoughts.